characters-program/programs/characters-program/src/state.rs

```diff
@@ -14,34 +16,43 @@ pub struct CharacterAccount {
 
 impl CharacterAccount {
     pub fn new(name: &str, authority: Pubkey, nft_address: Pubkey, owner: Pubkey) -> Result<Self> {
-        require!(
-            name.len() <= NAME_MAX_LENGTH,
-            ErrorCode::MaxNameLengthExceeded
-        );
-
-        let mut new_name = [0u8; NAME_MAX_LENGTH];
-        new_name[..name.len()].copy_from_slice(name.as_ref());
-
-        Ok(Self {
+        let mut account = Self {
             exp: 0,
-            name: new_name,
+            name: [0; NAME_MAX_LENGTH],
             authority,
             owner,
             nft_address,
-        })
+        };
+
+        account.set_name(name)?;
+
+        Ok(account)
     }
 
     pub fn update(&mut self, dto: UpdateCharacterDTO) -> ProgramResult {
         if let Some(name) = dto.name {
-            require!(
-                name.len() <= NAME_MAX_LENGTH,
-                ErrorCode::MaxNameLengthExceeded
-            );
-            self.name[..name.len()].copy_from_slice(name.as_ref());
+            self.set_name(&name)?;
         }
 
-        self.exp += dto.exp.unwrap_or(0);
+        self.set_exp(dto.exp.unwrap_or(0));
```

Comment:
I suggest changing the function name to `add_exp` since it's increasing the exp
```suggestion
        self.add_exp(dto.exp.unwrap_or(0));
```